src/GroupManager.php

```diff
@@ -123,10 +229,39 @@ public function removeGroup($entity_type_id, $bundle_id) {
   }
 
   /**
+   * Refreshes the locally stored data.
+   *
+   * Call this after making a change to a relationship between a group type and
+   * a group content type.
+   */
+  public function refresh() {
+    $this->refreshGroupMap();
+    unset($this->groupRelationMap);
+  }
+
+  /**
    * Refreshes the groupMap property with currently configured groups.
    */
   protected function refreshGroupMap() {
-    $this->groupMap = $this->configFactory->get(static::SETTINGS_CONFIG_KEY)->get(static::GROUPS_CONFIG_KEY);
+    $group_map = $this->configFactory->get(static::SETTINGS_CONFIG_KEY)->get(static::GROUPS_CONFIG_KEY);
+    $this->groupMap = !empty($group_map) ? $group_map : [];
+  }
+
+  /**
+   * Builds the map of relations between group types and group content types.
+   */
+  protected function refreshGroupRelationMap() {
+    $this->groupRelationMap = [];
```

Comment:
That is actually a really good idea. This has to loop over all fields in all bundles to find the group audience fields so this can become expensive on sites with lots of fields. This changes rarely so it makes sense to cache it.

It's not really config because this can be recalculated on the fly, shall we store it in State?